build.proj

```diff
@@ -2,61 +2,88 @@
+  <Target Name="Restore" DependsOnTargets="CreateOrUpdateCurrentVersionFile">
+    <ItemGroup>
+      <_RestoreProjects Include="external\dir.proj" />
+    </ItemGroup>
 
-  <PropertyGroup>
-    <PrebuildSetupTargets>
-      GenerateConfigurationProperties;
-      CreateOrUpdateCurrentVersionFile
-    </PrebuildSetupTargets>
-  </PropertyGroup>
+    <!-- Generate the configurations for this repo -->
+    <GenerateConfigurationProps Properties="@(Property)" PropertyValues="@(PropertyValue)" PropsFolder="$(BuildConfigurationFolder)" />
 
-  <PropertyGroup>
-    <TraversalBuildDependsOn>
-      $(PrebuildSetupTargets);
-      $(TraversalBuildDependsOn);
-    </TraversalBuildDependsOn>
-  </PropertyGroup>
+    <MSBuild Projects="@(_RestoreProjects)" Properties="$(ProjectProperties)" />
+  </Target>
 
-  <!-- Generate the configurations for this repo -->
-  <UsingTask TaskName="GenerateConfigurationProps" AssemblyFile="$(BuildToolsTaskDir)Microsoft.DotNet.Build.Tasks.dll"/>
-  <Target Name="GenerateConfigurationProperties">
-    <GenerateConfigurationProps Properties="@(Property)" PropertyValues="@(PropertyValue)" PropsFolder="$(BuildConfigurationFolder)" />
+  <Target Name="BuildNative">
+    <ItemGroup>
+      <_BuildNativeProjects Include="src\Native\build-native.proj" />
+    </ItemGroup>
+
+    <MSBuild Projects="@(_BuildNativeProjects)" Properties="$(ProjectProperties)" />
+  </Target>
+
+  <Target Name="BuildManaged">
+    <ItemGroup>
+      <_BuildMangedProjects Include="src\dirs.proj" />
+    </ItemGroup>
+
+    <MSBuild Projects="@(_BuildMangedProjects)" Properties="$(ProjectProperties)" />
   </Target>
 
-  <Target Name="Sync" DependsOnTargets="$(PrebuildSetupTargets)">
+  <Target Name="Test">
     <ItemGroup>
-      <ExternalProject Include="external\dir.proj" />
+      <_TestProjects Include="src\tests.builds" />
     </ItemGroup>
 
-    <MSBuild Projects="@(ExternalProject)"
-             ContinueOnError="ErrorAndContinue" />
+    <MSBuild Projects="@(_TestProjects)" Properties="$(ProjectProperties)" />
   </Target>
 
+  <Target Name="Pack">
+    <ItemGroup>
+      <!-- signing needs to happen before packaging -->
+      <_PackProjects Include="src\sign.proj" />
+      <_PackProjects Include="src\packages.builds" Condition="'$(BuildPackages)'=='true'" />
+    </ItemGroup>
+
+    <MSBuild Projects="@(_PackProjects)" Properties="$(ProjectProperties)" />
+  </Target>
+
+  <Import Project="$(ToolsDir)clean.targets" />
+
+  <Target Name="Build" DependsOnTargets="$(BuildDependsOn)" />
+
+  <Target Name="Clean">
+    <RemoveDir Directories="$(BinDir)" />
+    <!-- Temporarily outside BinDir -->
+    <RemoveDir Directories="$(PackageInstallPath)" />
+  </Target>
+
+  <Target Name="Rebuild" DependsOnTargets="Clean;Build" />
+
+  <!-- Hook that can be used to insert custom build tasks to the build process such as setup and/or cleanup tasks -->
+  <Import Project="build.override.targets" Condition="Exists('build.override.targets')" />
```

Comment:
Is this ever used?

Reply:
It was added as an extension point for wcf so they could just copy build.proj and still override some stuff. Beyond that I don't think it is used.